compiler/src/dotty/tools/dotc/transform/PostTyper.scala

```diff
@@ -261,9 +261,11 @@ class PostTyper extends MacroTransform with IdentityDenotTransformer { thisPhase
       def check(qual: Tree) =
         if !qual.tpe.isStable then
           report.error(em"Parameter untupling cannot be used for call-by-name parameters", tree.srcPos)
-      tree match
-        case Select(qual, _) => check(qual)    // simple select _n
-        case Apply(TypeApply(Select(qual, _), _), _) => check(qual) // generic select .apply[T](n)
+      appliedCore(closureBody(tree)) match
```

Comment:
Looks like a great fix. Since `appliedCore` is now used from outside `TreeInfo`, it would be good to give it a doc comment.